app/positiondirection.h

```diff
@@ -0,0 +1,58 @@
+#ifndef POSITIONDIRECTION_H
+#define POSITIONDIRECTION_H
+
+#include <QObject>
+#include <QTimer>
+
+#ifdef MOBILE_OS
+#include <QOrientationSensor>
+#include <QCompass>
+#endif
+
+#include "qgsquickpositionkit.h"
+
+class PositionDirection : public QObject
```

Comment:
Some basic comment for the class would be great (e.g. that it uses speed from GPS with higher speed or from compass+orientation when available, that it may have no valid direction at all, that it uses a minimal angle for updates of direction)